packages/x-components/src/x-modules/search-box/components/__tests__/search-input-placeholder.spec.ts

```diff
@@ -0,0 +1,188 @@
+    wrapper,
+    getPlaceholderText,
+    isPlaceholderVisible,
+    hoverInput,
+    emit
+  };
+}
+
+interface MountTestComponentOptions {
+  messages?: Array<string>;
+  animationIntervalMs?: number;
+  animateOnlyOnHover?: boolean;
+  autofocus?: boolean;
+}
+
+interface MountTestComponentAPI {
+  messages: Array<string>;
+  wrapper: Wrapper<Vue>;
+  getPlaceholderText: () => string;
+  isPlaceholderVisible: () => boolean;
+  hoverInput: (mode: 'in' | 'out') => Promise<void>;
+  emit: (event: XEvent, payload: any) => Promise<void>;
+}
+
+describe('testing search input placeholder component', () => {
+  beforeAll(jest.useFakeTimers);
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  afterEach(jest.clearAllTimers);
+
+  it('is an XComponent', async () => {
+    const { wrapper } = await mountTestComponent();
+    const placeholderWrapper = wrapper.findComponent(SearchInputPlaceholder);
+
+    expect(isXComponent(placeholderWrapper.vm)).toEqual(true);
+  });
+
+  it('has SearchBox as XModule', async () => {
+    const { wrapper } = await mountTestComponent();
+    const placeholderWrapper = wrapper.findComponent(SearchInputPlaceholder);
+
+    expect(getXComponentXModuleName(placeholderWrapper.vm)).toEqual('searchBox');
+  });
+
+  it('animates the rendered text always', async () => {
+    const { messages, wrapper, getPlaceholderText } = await mountTestComponent();
+
+    for (const message of messages) {
+      expect(getPlaceholderText()).toEqual(message);
+      jest.runOnlyPendingTimers();
+      await wrapper.vm.$nextTick();
+    }
+    expect(getPlaceholderText()).toEqual(messages[0]);
+  });
+
+  it('animates the rendered text only on hover if configured to do so', async () => {
+    const { messages, wrapper, getPlaceholderText, hoverInput } = await mountTestComponent({
+      animateOnlyOnHover: true
+    });
+
+    expect(getPlaceholderText()).toEqual(messages[0]);
+    await hoverInput('in');
+    for (const message of messages.slice(1)) {
+      expect(getPlaceholderText()).toEqual(message);
+      jest.runOnlyPendingTimers();
+      await wrapper.vm.$nextTick();
+    }
+    expect(getPlaceholderText()).toEqual(messages[0]);
+    jest.runOnlyPendingTimers();
+    await wrapper.vm.$nextTick();
+    expect(getPlaceholderText()).toEqual(messages[1]);
+    await hoverInput('out');
+    expect(getPlaceholderText()).toEqual(messages[0]);
+    jest.advanceTimersByTime(2000);
+    expect(getPlaceholderText()).toEqual(messages[0]);
+    await hoverInput('in');
+    expect(getPlaceholderText()).toEqual(messages[1]);
+  });
+
+  it('is not visible when there is a query set', async () => {
+    const { messages, getPlaceholderText, isPlaceholderVisible, emit } = await mountTestComponent();
+
+    expect(getPlaceholderText()).toEqual(messages[0]);
+    await emit('UserAcceptedAQuery', 'testing');
+    expect(isPlaceholderVisible()).toEqual(false);
+    jest.advanceTimersByTime(2000);
+    await emit('UserAcceptedAQuery', undefined);
```

Comment:
UserAcceptedAQuery expects a string

```suggestion
    await emit('UserAcceptedAQuery', '');
```